src/api/article/controller.js

```diff
@@ -61,67 +64,173 @@ export const getOne = ({ params: { slug }, user }, res, next) =>
-    }
-    sendJson(res, code)(data);
-  } catch (err) {
-    next(err);
-  }
-};
+export const create = ({ body }, res, next) =>
+  // TODO(uladbohdan): to deprecate body.brand.
+  ArticleBrand.findOne({ slug: body.brand || body.brandSlug })
+    .then(checkIsFound) // Brand is required.
+    .then(({ _id: brandId }) =>
+      ArticleCollection.findOne({ slug: body.collectionSlug })
+        .then(collection => collection && collection._id) // Collection is not required.
+        .then(collectionId =>
+          User.findOne({ email: body.authorEmail, role: 'author' })
+            .then(author => author && author._id) // Author is not required.
+            .then(authorId =>
+              Article({
+                ...omit(body, ['locales']),
+                author: authorId,
+                brand: brandId,
+                collectionId,
+              })
+                .save()
+                .then(async article => {
+                  // Processing with localizations (Bundled API).
+                  if (body.locales) {
+                    await Promise.all(
+                      Object.values(body.locales).map(localization =>
+                        LocalizedArticle({
+                          ...localization,
+                          articleId: article._id,
+                        })
+                          .save()
+                          .then(({ _id: locId }) => article.locales.push(locId))
+                          .catch(next)
+                      )
+                    );
+                  }
+                  return article;
+                })
+                .then(article => article.save())
+                .then(({ _id: articleId }) =>
+                  Article.findOne({ _id: articleId })
+                    .populate('author', POPULATE_OPTIONS.author)
+                    .populate('brand', POPULATE_OPTIONS.brand)
+                    .populate('collectionId', POPULATE_OPTIONS.collection)
+                    .populate('locales', POPULATE_OPTIONS.locales)
+                    .then(serializeArticle)
+                    .then(sendJson(res))
+                    .catch(next)
+                )
+                .catch(next)
+            )
+            .catch(next)
+        )
+        .catch(next)
+    )
+    .catch(next);
 
-export const update = ({ params: { slug }, body }, res, next) =>
-  LocalizedArticle.findOne({ slug })
+export const update = ({ params: { slugOrId }, body }, res, next) =>
+  LocalizedArticle.findOne({ slug: slugOrId, active: true })
+    .then(result => (result && result.articleId) || (isValidId(slugOrId) && slugOrId))
     .then(checkIsFound)
-    .then(({ articleId }) =>
-      Article.findOneAndUpdate({ _id: articleId }, body, { new: true })
-        .populate('author', POPULATE_OPTIONS.author)
-        .populate('brand', POPULATE_OPTIONS.brand)
-        .populate('collectionId', POPULATE_OPTIONS.collection)
-        .populate('locales', POPULATE_OPTIONS.locales)
```

Comment:
it was such a small & beautiful method 😭

Reply:
it was indeed :(